mesonbuild/backend/ninjabackend.py

```diff
@@ -845,6 +850,8 @@ def generate_target(self, target):
             compiled_sources.append(s)
             source2object[s] = o
             obj_list.append(o)
+            if s.endswith('.d'):
```

Comment:
Is there a predefined routine to identify the language of a source file?

Reply:
https://github.com/mesonbuild/meson/blob/29c26d5b26397eaa3606d22d71309ecd1eb6b223/mesonbuild/compilers/compilers.py#L54-L61

It's not a routine, but it is a predefined lookup table.

Reply:
or you could do `s.split('.')[1] in compilers.lang_suffixes['d']`

Reply:
yes OK. I used `s.split('.')[-1]` as there could be another point in the path